tests/components/vacuum/test_init.py

```diff
@@ -613,3 +613,37 @@ async def test_vacuum_log_deprecated_battery_supported_feature(
         ", please report it to the author of the 'test' custom integration"
         in caplog.text
     )
+
+
+async def test_vacuum_not_log_deprecated_battery_properties_during_init(
+    hass: HomeAssistant,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Test not logging deprecation until after added to hass."""
+
+    class MockLegacyVacuum(MockVacuum):
+        """Mocked vacuum entity."""
+
+        def __init__(self, **kwargs: Any) -> None:
+            """Initialize a mock vacuum entity."""
+            super().__init__(**kwargs)
+            self._attr_battery_level = 50
+
+        @property
+        def activity(self) -> str:
+            """Return the state of the entity."""
+            return VacuumActivity.CLEANING
+
+    entity = MockLegacyVacuum(
+        name="Testing",
+        entity_id="vacuum.test",
+    )
+    assert entity.battery_level == 50
+
+    assert (
+        "Detected that custom integration 'test' is setting the battery_level which has been deprecated."
```

Comment:
I suggest we just check the first sentence isn't in the log. The longer the sentence the easier it is to make a mistake. Since we're checking that it's not in the log a typo will not fail the tests.